corehq/apps/reports/util.py

```diff
@@ -851,3 +852,34 @@ def domain_copied_cases_by_owner(domain, owner_ids):
         .owner(owner_ids)\
         .NOT(case_property_missing(CaseCopier.COMMCARE_CASE_COPY_PROPERTY_NAME))\
         .values_list('_id', flat=True)
+
+
+def get_commcare_version_and_date_from_last_usage(last_submission=None, last_device=None, formatted=False):
+    """
+    Gets CommCare version and date from the last submission, or fall back to the last used device.
+
+    Args:
+    last_submission: Dictionary containing Last Submission data from ES
+    last_device: Dictionary containing DeviceIdLastUsed data from ES
+    """
+    from corehq.apps.reports.standard.deployments import format_commcare_version
+
+    version = None
+    version_in_use = None
+    date_of_use = None
+
+    if last_submission and last_submission.get('commcare_version'):
+        version = last_submission.get('commcare_version')
+        date_of_use = last_submission.get('submission_date')
+
+    elif last_device and last_device.get('commcare_version'):
+        version = last_device.get('commcare_version')
+        date_of_use = last_device.get('last_used')
+
+    if version:
+        version_in_use = format_commcare_version(version) if formatted else version
+
+    if isinstance(date_of_use, str):
+        date_of_use = datetime.strptime(date_of_use, ISO_DATETIME_FORMAT)
```

Comment:
Is this check necessary? I think `date_of_use` will always be a string here.  I think the conditional suggests that sometimes it won't be a string, which is confusing if that isn't true.

Reply:
82bb7616018d97ba7eec526d941a0595f20cdf53